src/components/Clock.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+interface State {
+  today: Date;
+}
+
+interface ClockProps {
+  clockName: string;
+}
+
+export class Clock extends React.Component<ClockProps, State> {
+  state = {
+    today: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ today: new Date() });
+      // eslint-disable-next-line no-console
+      console.info((this.state.today).toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Readonly<ClockProps>) {
+    const { clockName } = this.props;
+
+    if (prevProps.clockName !== clockName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.clockName} to ${clockName}`);
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timerId);
+  }
+
+  render() {
+    const { clockName } = this.props;
+    const { today } = this.state;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {today.toUTCString().slice(-12, -4)}
```

Comment:
please don't call a function from the template because it will be executed every time when the state change, so make sense to move it to the variable